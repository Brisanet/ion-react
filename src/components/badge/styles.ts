import styled, { DefaultTheme, RuleSet, css } from 'styled-components';
import { BadgeType } from './badge';

type BadgeStylesProps = {
  $type: BadgeType;
};

type ColorByVariant = {
  background: string;
  color?: string;
};

export const typeColors: (
  theme: DefaultTheme,
  type: BadgeType
) => ColorByVariant = ({ colors }, type) =>
  ({
    primary: {
      background: colors.main.primary,
    },
    secondary: {
      background: colors.primary[1],
      color: colors.main.primary,
    },
    neutral: {
      background: colors.neutral[4],
      color: colors.neutral[8],
    },
    danger: {
      background: colors.main.negative,
    },
  }[type]);

const type: (theme: DefaultTheme, type: BadgeType) => RuleSet<object> = (
  theme,
  type
) => {
  const { background, color } = typeColors(theme, type);

  return css`
    background-color: ${background};
    color: ${color || theme.colors.neutral[1]};
  `;
};

export const Badge = styled.span<BadgeStylesProps>`
  ${({ theme, $type }) => css`
    ${theme.font.size[12]};
    font-weight: 600;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    padding: 0 0.4rem;
    border-radius: 50rem;
    user-select: none;
    ${type(theme, $type)};
  `}
`;
