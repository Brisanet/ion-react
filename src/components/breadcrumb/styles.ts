import stitches from '../../stitches.config';
import { spacing } from '../utils/spacing';
const { styled } = stitches;

export const BreadcrumbStyles = styled('nav', {
  padding: `${spacing(0.6)} ${spacing(0.9)}`,
  textDecoration: 'none',
  fontFamily: 'sans-serif',
  fontSize: '15px',

  '& > ol': {
    display: 'inline-flex',
    margin: '0',
    padding: `${spacing(1.25)}`,

    '& li': {
      display: 'flex',
      alignItems: 'center',
      textAlign: 'center',

      svg: {
        fill: '$neutral6',
      },

      a: {
        padding: `${spacing(1.25)}`,
        textDecoration: 'none',
        color: '$neutral6',
        fontFamily: 'sans-serif',
        fontSize: '14px',
        fontWeight: '400',
        cursor: 'pointer',
        '&:hover': {
          color: '$primary6',
        },
        '&:active': {
          fontWeight: '600',
          fontSize: '14px',
          color: '$primary6',
        },
      },

      '&:last-child': {
        a: {
          color: '$primary6',
          fontWeight: '600',
          cursor: 'default',
        },
        svg: { display: 'none' },
      },
    },
  },
});
