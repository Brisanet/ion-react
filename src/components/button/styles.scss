.btn {
  font-size: 30px;
  padding: 10px 20px;
  background-color: blueviolet;
}
