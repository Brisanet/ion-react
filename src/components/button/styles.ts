import stitches from '../../stitches.config';
import { getBgFontBorderColors } from '../utils/getBgFontBorderColors';
import { getButtonSize } from '../utils/getButtonSize';

const { styled } = stitches;

// Componente Button
export const ButtonStyles = styled('button', {
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: '8px',
  fontWeight: '600',
  cursor: 'pointer',
  gap: '8px',

  variants: {
    type: {
      primary: {
        ...getBgFontBorderColors({
          background: '$primaryColor',
          color: '$neutral1',
        }),

        '&:hover': {
          background: '$primary5',
        },
        '&:active': { background: '$primary7' },
        '&:disabled': {
          ...getBgFontBorderColors({
            background: '$neutral2',
            color: '$neutral5',
          }),
          cursor: 'not-allowed',
        },
      },
      secondary: {
        ...getBgFontBorderColors({
          background: '$neutral1',
          color: '$primaryColor',
          border: '1px solid $neutral4',
        }),

        '&:hover': getBgFontBorderColors({
          background: '$primary1',
          color: '$primary5',
          border: '1px solid $primary4',
        }),
        '&:active': getBgFontBorderColors({
          background: '$primary2',
          color: '$primary5',
          border: '1px solid $primary7',
        }),
        '&:disabled': {
          ...getBgFontBorderColors({
            background: '$neutral2 ',
            color: '$neutral5 ',
            border: '1px solid $neutral5 ',
          }),
          cursor: 'not-allowed',
        },
      },
      ghost: {
        ...getBgFontBorderColors({
          background: 'transparent',
          color: '$primaryColor',
        }),

        '&:hover': {
          ...getBgFontBorderColors({
            background: '$primary1',
            color: '$primary5',
          }),
        },
        '&:active': {
          ...getBgFontBorderColors({
            background: '$primary2',
            color: '$primary7',
          }),
        },
        '&:disabled': {
          ...getBgFontBorderColors({
            background: 'transparent',
            color: '$neutral5',
          }),
          cursor: 'not-allowed',
        },
      },
      dashed: {
        ...getBgFontBorderColors({
          background: 'transparent',
          color: '$primary7',
          border: '1px dashed $neutral4',
        }),

        '&:hover': {
          ...getBgFontBorderColors({
            background: '$primary1',
            color: '$primary5',
            border: '1px dashed $primary4',
          }),
        },
        '&:active': {
          ...getBgFontBorderColors({
            background: '$primary2',
            color: '$primary7',
            border: '1px dashed $primary5',
          }),
        },
        '&:disabled': {
          ...getBgFontBorderColors({
            background: '$neutral3',
            color: '$neutral5',
            border: '1px dashed $neutral5',
          }),
          cursor: 'not-allowed',
        },
      },
    },

    danger: {
      primary: {
        ...getBgFontBorderColors({
          background: '$negative6',
          color: '$negative1',
        }),
        '&:hover': {
          ...getBgFontBorderColors({
            background: '$negative5',
            color: '$neutral1',
          }),
        },
        '&:active': {
          background: '$negative7',
        },
        '&:disabled': {
          ...getBgFontBorderColors({
            background: '$neutral2',
            color: '$neutral5',
          }),
          cursor: 'not-allowed',
        },
      },
      secondary: {
        color: '$negativeColor',
        '&:hover': {
          ...getBgFontBorderColors({
            background: '$negative1',
            color: '$negative5',
            border: '1px solid $negative3',
          }),
        },
        '&:active': {
          ...getBgFontBorderColors({
            background: '$negative2',
            color: '$negative7',
            border: '1px solid $negative7',
          }),
        },
        '&:disabled': {
          ...getBgFontBorderColors({
            background: '$neutral2 ',
            color: '$neutral5 ',
            border: '1px solid $neutral5',
          }),
          cursor: 'not-allowed',
        },
      },
      ghost: {
        color: '$negative6',
        '&:hover': {
          ...getBgFontBorderColors({
            background: '$negative1',
            color: '$negative5',
          }),
        },
        '&:active': {
          ...getBgFontBorderColors({
            background: '$negative2',
            color: '$negative7',
          }),
        },
        '&:disabled': {
          ...getBgFontBorderColors({
            background: 'transparent',
            color: '$neutral5',
          }),
          cursor: 'not-allowed',
        },
      },
      dashed: {
        border: '1px dashed $neutral4',
        color: '$negative6',
        '&:hover': {
          ...getBgFontBorderColors({
            background: '$negative1',
            color: '$negative5',
            border: '1px dashed $negative3',
          }),
        },
        '&:active': {
          ...getBgFontBorderColors({
            background: '$negative2',
            color: '$negative7',
            border: '1px dashed $negative6',
          }),
        },
        '&:disabled': {
          ...getBgFontBorderColors({
            background: '$neutral3',
            color: '$neutral5',
            border: '1px dashed $neutral5',
          }),
          cursor: 'not-allowed',
        },
      },
      none: {},
    },

    size: {
      sm: {
        ...getButtonSize({
          buttonSize: '24px',
          padding: '4px 12px',
          fontSize: '12px',
        }),
      },
      md: {
        ...getButtonSize({
          buttonSize: '32px',
          padding: '6px 16px',
          fontSize: '14px',
        }),
      },
      lg: {
        ...getButtonSize({
          buttonSize: '40px',
          padding: '8px 20px',
          fontSize: '16px',
        }),
      },
      xl: {
        ...getButtonSize({
          buttonSize: '48px',
          padding: '12px 24px',
          fontSize: '16px',
        }),
      },
    },
  },
});
