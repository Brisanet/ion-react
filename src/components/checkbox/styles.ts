import stitches from '../../stitches.config';

const { styled } = stitches;

export const CheckboxStyles = styled('div', {});
