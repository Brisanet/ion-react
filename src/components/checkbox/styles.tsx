import { css, styled } from 'styled-components';

const checkedIconBackground = (color: string) => css`
  ${() => {
    const encodedColor = encodeURIComponent(color);
    return css`
      background-image: ${`url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='${encodedColor}' xmlns='http://www.w3.org/2000/svg'%3E%3Crect width='16' height='16' rx='4' fill='none'/%3E%3Cpath d='M12.8839 6.28075C13.3721 5.79258 13.3721 5.00113 12.8839 4.51298C12.3957 4.02483 11.6043 4.02484 11.1161 4.513L6.7939 8.83535L4.88385 6.9253C4.3957 6.43714 3.60424 6.43714 3.11609 6.9253C2.62793 7.41345 2.62793 8.20491 3.11609 8.69307L5.91003 11.487C6.14445 11.7214 6.4624 11.8531 6.79392 11.8531C7.12545 11.8531 7.44339 11.7214 7.67781 11.487L12.8839 6.28075Z' fill='%23${encodedColor}'/%3E%3C/svg%3E%0A")`};
      background-size: cover;
    `;
  }};
`;

export const CheckboxContainer = styled.div`
  display: inline-flex;
  align-items: center;
  gap: 0.8rem;
`;

export const Checkbox = styled.input.attrs({ type: 'checkbox' })`
  ${({ theme }) => css`
    appearance: none;
    width: 1.6rem;
    height: 1.6rem;
    margin: 0.4rem;
    border-radius: 0.4rem;
    border: 1px solid ${theme.colors.neutral[5]};
    background: ${theme.colors.neutral[1]};
    cursor: pointer;
    ${theme.utils.focus}

    &:enabled {
      &:hover, &:focus-visible {
        box-shadow: 0 0 0.2rem 0.2rem ${theme.colors.primary[2]};
        border-color: ${theme.colors.primary[3]};
        ${checkedIconBackground(theme.colors.primary[2])};
      }

      &:active {
        box-shadow: none;
        background-color: ${theme.colors.primary[2]};
        border-color: ${theme.colors.primary[5]};
        ${checkedIconBackground(theme.colors.primary[1])};
      }

      &:checked {
        border: none;
        background-color: ${theme.colors.primary[6]};
        ${checkedIconBackground(theme.colors.primary[1])};

        &:hover {
          background-color: ${theme.colors.primary[5]};
        }

        &:active {
          background-color: ${theme.colors.primary[7]};
        }
      }
    }

    &:disabled {
      background-color: ${theme.colors.neutral[3]};
      border: 1px solid ${theme.colors.neutral[4]};
      cursor: not-allowed;

      &:checked {
        background-color: ${theme.colors.neutral[2]};
        border: none;
        ${checkedIconBackground(theme.colors.neutral[4])};
      }
    },
  `}
`;

export const Label = styled.label`
  ${({ theme }) => css`
    ${theme.font.size[14]};
    color: ${theme.colors.neutral[7]};

    ${Checkbox}:disabled + & {
      color: ${theme.colors.neutral[5]};
    }
  `};
`;
