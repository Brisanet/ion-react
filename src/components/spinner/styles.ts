import { css, DefaultTheme, keyframes, styled } from 'styled-components';
import { IonSpinnerProps } from './spinner';

type SpinnerStylesProps = {
  $variant?: IonSpinnerProps['variant'];
  $size?: IonSpinnerProps['size'];
  $customColor?: IonSpinnerProps['customColor'];
};

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const colorsByType = (
  theme: DefaultTheme,
  type: SpinnerStylesProps['$variant'] = 'primary'
) =>
  ({
    primary: theme.colors.main.primary,
    secondary: theme.colors.main.secondary,
    danger: theme.colors.main.negative,
  }[type]);

export const Container = styled.span``;

export const Spinner = styled.span<SpinnerStylesProps>`
  ${({ theme, $variant, $size, $customColor }) => css`
    width: ${$size}px;
    height: ${$size}px;
    display: block;
    border: 2px solid transparent;
    border-left-color: ${!$customColor
      ? colorsByType(theme, $variant)
      : $customColor};
    border-radius: 50%;
    animation: ${spin} 1s linear infinite;
  `}
`;
