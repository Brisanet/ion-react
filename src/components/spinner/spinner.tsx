import { validateHexColor } from '../utils/validateHexColor';
import { Container, Spinner } from './styles';

type Variant = 'primary' | 'secondary' | 'danger';

export interface IonSpinnerProps {
  variant?: Variant;
  size?: number;
  customColor?: string;
}

export const IonSpinner = ({
  variant = 'primary',
  size = 32,
  customColor = '',
}: IonSpinnerProps) => {
  return (
    <Container>
      <Spinner
        data-testid='ion-spinner'
        $variant={variant}
        $size={size}
        $customColor={validateHexColor(customColor) ? customColor : undefined}
      />
    </Container>
  );
};
