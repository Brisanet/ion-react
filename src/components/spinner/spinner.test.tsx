import theme from '@ion/styles/theme';
import { screen } from '@testing-library/dom';
import { renderWithTheme } from '../utils/test-utils';
import { IonSpinner, IonSpinnerProps } from './spinner';
import { colorsByType } from './styles';

const variants: IonSpinnerProps['variant'][] = [
  'primary',
  'secondary',
  'danger',
];

const sut = (props?: IonSpinnerProps) => {
  return renderWithTheme(<IonSpinner {...props} />);
};

describe('Spinner', () => {
  it('should render spinner with default props', () => {
    const { container } = sut();
    expect(screen.getByTestId('ion-spinner')).toHaveStyleRule(
      'border-left-color',
      colorsByType(theme)
    );
    expect(container).toMatchSnapshot();
  });
  describe('Variants', () => {
    it.each(variants)('should render spinner with %s variant', (variant) => {
      const { container } = sut({ variant });
      expect(screen.getByTestId('ion-spinner')).toHaveStyleRule(
        'border-left-color',
        colorsByType(theme, variant)
      );
      expect(container).toMatchSnapshot();
    });
  });
  describe('Size', () => {
    it('should render spinner with 64px size', () => {
      const { container } = sut({ size: 64 });
      const spinner = screen.getByTestId('ion-spinner');
      expect(spinner).toHaveStyleRule('width', '64px');
      expect(spinner).toHaveStyleRule('height', '64px');
      expect(container).toMatchSnapshot();
    });
  });
  describe('Custom color', () => {
    it('should render spinner with custom color', () => {
      const { container } = sut({ customColor: '#f00' });
      expect(screen.getByTestId('ion-spinner')).toHaveStyleRule(
        'border-left-color',
        '#f00'
      );
      expect(container).toMatchSnapshot();
    });
  });
});
