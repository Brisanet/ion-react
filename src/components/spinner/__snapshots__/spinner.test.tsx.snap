// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Custom color should render spinner with custom color 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  display: block;
  border: 2px solid transparent;
  border-left-color: #f00;
  border-radius: 50%;
  animation: bKPVWD 1s linear infinite;
}

<div>
  <span
    class=""
  >
    <span
      class="c0"
      data-testid="ion-spinner"
    />
  </span>
</div>
`;

exports[`Spinner Size should render spinner with 64px size 1`] = `
.c0 {
  width: 64px;
  height: 64px;
  display: block;
  border: 2px solid transparent;
  border-left-color: #0858ce;
  border-radius: 50%;
  animation: bKPVWD 1s linear infinite;
}

<div>
  <span
    class=""
  >
    <span
      class="c0"
      data-testid="ion-spinner"
    />
  </span>
</div>
`;

exports[`Spinner Variants should render spinner with danger variant 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  display: block;
  border: 2px solid transparent;
  border-left-color: #d6293a;
  border-radius: 50%;
  animation: bKPVWD 1s linear infinite;
}

<div>
  <span
    class=""
  >
    <span
      class="c0"
      data-testid="ion-spinner"
    />
  </span>
</div>
`;

exports[`Spinner Variants should render spinner with primary variant 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  display: block;
  border: 2px solid transparent;
  border-left-color: #0858ce;
  border-radius: 50%;
  animation: bKPVWD 1s linear infinite;
}

<div>
  <span
    class=""
  >
    <span
      class="c0"
      data-testid="ion-spinner"
    />
  </span>
</div>
`;

exports[`Spinner Variants should render spinner with secondary variant 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  display: block;
  border: 2px solid transparent;
  border-left-color: #ebf3fe;
  border-radius: 50%;
  animation: bKPVWD 1s linear infinite;
}

<div>
  <span
    class=""
  >
    <span
      class="c0"
      data-testid="ion-spinner"
    />
  </span>
</div>
`;

exports[`Spinner should render spinner with default props 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  display: block;
  border: 2px solid transparent;
  border-left-color: #0858ce;
  border-radius: 50%;
  animation: bKPVWD 1s linear infinite;
}

<div>
  <span
    class=""
  >
    <span
      class="c0"
      data-testid="ion-spinner"
    />
  </span>
</div>
`;
